packages/common/infra/src/livedata/index.ts

```diff
@@ -170,6 +170,67 @@ export class LiveData<T = unknown> implements InteropObservable<T> {
     );
   }
 
+  /**
+   * @example
+   * ```ts
+   * const a = new LiveData('v1');
+   * const v1 = new LiveData(100);
+   * const v2 = new LiveData(200);
+   *
+   * const v = LiveData.computed(get => {
+   *   return get(a) === 'v1' ? get(v1) : get(v2);
+   * });
+   *
+   * expect(v.value).toBe(100);
+   * ```
+   */
+  static computed<T>(
+    compute: (get: <L>(data: LiveData<L>) => L) => T
+  ): LiveData<T> {
+    return LiveData.from(
+      new Observable(subscribe => {
+        const execute = (next: () => void) => {
+          const subscriptions: Subscription[] = [];
+          const getfn = <L>(data: LiveData<L>) => {
+            let value = null as L;
+            let first = true;
+            subscriptions.push(
```

Comment:
Can we deduplicate the same deps?

Reply:
Subscribing to the same livedata multiple times actually shares the same subscription internally, so it is safe.

Reply:
see this test
https://github.com/toeverything/AFFiNE/pull/6091/files#diff-8260747d906aadc2441485759654a60aac8a7d00449af79ddbbb61b77307f180R286